src/parser/mod.rs

```diff
@@ -25,103 +18,42 @@ pub use crate::parser::utils::FastxReader;
 // TODO: for now this drops support for stdin parsing. It could be added back if needed though
 // by adding a method to the readers to set some initial buffer.
 
-fn read_file(mut f: File) -> Result<Box<dyn FastxReader>, ParseError> {
-    let mut first = [0; 1];
-    f.read_exact(&mut first)?;
-    // Back to the beginning of the file
-    f.seek(SeekFrom::Start(0))?;
-
-    match first[0] {
-        b'>' => Ok(Box::new(FastaReader::new(f))),
-        b'@' => Ok(Box::new(FastqReader::new(f))),
-        _ => Err(ParseError::new_unknown_format(first[0])),
-    }
-}
-
 /// The main entry point of needletail.
 /// Parses the file given a path and return an iterator-like reader struct.
 /// This automatically detects whether the file is:
 /// 1. compressed: gzip, bz and xz are supported and will use the appropriate decoder
 /// 2. FASTA or FASTQ: the right parser will be automatically instantiated
 /// 1 is only available if the `compression` feature is enabled.
-#[cfg(not(feature = "compression"))]
 pub fn parse_fastx_file<P: AsRef<Path>>(path: P) -> Result<Box<dyn FastxReader>, ParseError> {
-    let f = File::open(&path)?;
-    read_file(f)
+    let filename = path.as_ref();
+    let f: Box<dyn Read> = if filename == Path::new("-") {
+        Box::new(stdin())
+    } else {
+        Box::new(File::open(path)?)
+    };
+    parse_fastx_reader(f)
 }
 
-// Magic bytes for each compression format
-#[cfg(feature = "compression")]
-const GZ_MAGIC: [u8; 2] = [0x1F, 0x8B];
-#[cfg(feature = "compression")]
-const BZ_MAGIC: [u8; 2] = [0x42, 0x5A];
-#[cfg(feature = "compression")]
-const XZ_MAGIC: [u8; 2] = [0xFD, 0x37];
-
-/// The main entry point of needletail.
-/// Parses the file given a path and return an iterator-like reader struct.
-/// This automatically detects whether the file is:
+/// Parses any input supporting the Read trait and return an iterator-like reader struct.
+/// This automatically detects whether the input is:
 /// 1. compressed: gzip, bz and xz are supported and will use the appropriate decoder
```

Comment:
I think xz is not supported anymore right?

Reply:
It is, but we exposed the feature in niffler as `lzma`